libs/natparse/src/main/java/org/amshove/natparse/lexing/Lexer.java

```diff
@@ -15,10 +17,13 @@
 	private int line;
 	private int currentLineStartOffset;
 	private Path filePath;
+	private NaturalHeader sourceHeader;
 	private IPosition relocatedDiagnosticPosition;
 
-	private boolean inParens;
+	private boolean inParens, inHdr, hdrDone;
```

Comment:
Can you make separate attributes (with `private boolean`) in separate lines for this?

Also `inSourceHeader` and `sourceHeaderDone` might be more descriptive names.